RoslynPluginGenerator/ZipExtensions.cs

```diff
@@ -58,4 +60,32 @@ public static void CreateFromDirectory(string sourceDirectoryName, string destin
             }
         }
     }
+
+    /// <summary>
+    /// Replaces the back slash to foward slash to attend the international convention of directories separation.
+    /// This will allow the zip file to be unzipped the correct way both linux and windows OS's.
+    /// </summary>
+    /// <seealso cref="System.Text.UTF8Encoding" />
+    class PathEncoder : UTF8Encoding
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PathEncoder"/> class.
+        /// </summary>
+        public PathEncoder() : base(true)
```

Comment:
Why is it needed for the GetPreamble to return BOM for encoding these file names?